modules/avm/main.tf

```
@@ -1,14 +1,13 @@
 locals {
-  email               = var.email != null ? var.email : "${local.prefixed_email}@schubergphilis.com"
-  name                = var.environment != null ? "${var.name}-${var.environment}" : var.name
-  prefixed_email      = "${var.defaults.account_prefix}-aws-${local.name}"
-  prefixed_name       = "${var.defaults.account_prefix}-${local.name}"
+  email               = var.email != null ? var.email : "${var.defaults.account_prefix}-aws-${local.name}@schubergphilis.com"
+  name                = var.environment == null || var.environment == "prod" ? var.name : "${var.name}-${var.environment}"
   organizational_unit = var.organizational_unit != null ? var.organizational_unit : var.environment == "prod" ? "Production" : "Non-Production"
+  prefixed_name       = "${var.defaults.account_prefix}-${local.name}"
 }
 
 module "account" {
   source                   = "github.com/schubergphilis/terraform-aws-mcaf-account?ref=v0.3.0"
-  account                  = local.name
+  account                  = local.prefixed_name
```

As discussed, let's revert these changes and add an `account_name` variable that will be used here if it's set:

```suggestion
  account                  = coalesce(var.account_name, local.prefixed_name)
```

Rest of the proposed changes can be reverted.

---

I've used `coalesce(var.account_name, local.name)` instead of `coalesce(var.account_name, local.prefixed_name)` to keep it backwards compatible and to also follow our discussion regarding prefixes not been really necessary.